nessie/system_test.go

```diff
@@ -0,0 +1,154 @@
+package nessie
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	genclient "github.com/treeverse/lakefs/api/gen/client"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/swag"
+	"github.com/thanhpk/randstr"
+	"github.com/treeverse/lakefs/api"
+	"github.com/treeverse/lakefs/api/gen/client/setup"
+	"github.com/treeverse/lakefs/api/gen/models"
+	"github.com/treeverse/lakefs/logging"
+)
+
+// testsConfig contains configuration needed for running the tests
+type testsConfig struct {
+	// baseURL is the base address of the lakeFS endpoint
+	baseURL string
+
+	// rawBucketPath is the full path to the s3 bucket
+	rawBucketPath string
+
+	enabled bool
+	https   bool
+}
+
+var (
```

Comment:
I am uncomfortable with these global variables.  E.g. if we nested all tests inside a single function (and used `t.Run`) they could be local.  E.g. if they were attached to a single object we could construct it and pass _that_ around.
Rationale: the usual "global variables are only convenient in the short term" argument.  Consider how this scales to add more tests.  Even `client` might be pointed at different clients each time.